### PR TITLE
Create endpoint to get users with pagination 

### PR DESCRIPTION
**Pull request by KsanderERRor**

### CODE REVIEW

File: server/api/user/user.service.js [hunk 1]
```diff
@@ -13,4 +13,32 @@ module.exports = {
...
+  getUserByID: async (userId) => {
+    return await User.findById(userId);
+  },
+
+  updateUserByID: async (userId,updateData) => {
+    return await User.findByIdAndUpdate(userId,updateData);
+  },
+
+  getAllUsersPagination: async ( query = {} ) => {
+    const { page = 1, perPage = 5 } = query;
```
Comment (chestnyh) [inline server/api/user/user.service.js:30]: If I set `perPage = 1 000 000` do you ask for 1000000 rows from db

File: server/commonMiddleware/common.middleware.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+      if (!user) {
+        res.json({ message: `User with id ${req.params.userId} is not defined` });
+      }
+
+      req.locals = {...req.locals, user };
+
+      next();
+    } catch (e) {
+      throw new Error(e);
```
Comment (chestnyh) [inline server/commonMiddleware/common.middleware.js:17]: Catch error to throw an error? What the reason?

File: server/api/user/user.controller.js [hunk 3]
```diff
@@ -5,10 +5,48 @@ module.exports = {
...
+
+      res.json(user);
+    } catch (e) {
+      next(e);
+    }
+  },
+
+  updateOneUserByID: async (req, res, next) => {
+    try {
+       const updatedUser = await userService.updateUserByID(req.params.userId, req.body);
```
Comment (chestnyh) [inline server/api/user/user.controller.js:36]: Where do you use `updatedUser` variable

File: server/api/user/user.controller.js [hunk 4]
```diff
@@ -5,10 +5,48 @@ module.exports = {
...
     }
   },
+
+  deleteUser: async (req, res, next) => {
+    try {
+      await userService.deleteUserByID(req.params.userId);
+
+      res.json(`user was delete`);
+    } catch (e) {
+      next(e);
```
Comment (chestnyh) [inline server/api/user/user.controller.js:20]: Can't understand where do you pass this error?

File: server/api/user/user.service.js (hunk 1 above)
Comment (KsanderERRor) [reply server/api/user/user.service.js:30]: I tried, it working

File: server/api/user/user.controller.js (hunk 3 above)
Comment (KsanderERRor) [reply server/api/user/user.controller.js:36]: I dont know, do I need sending updateUser() + message  or only message?

File: server/api/user/user.controller.js (hunk 4 above)
Comment (KsanderERRor) [reply server/api/user/user.controller.js:20]: then I planned to create error handler and pass errors there

**Review by chestnyh — APPROVED** [review]

**Pull request merged**